packages/blockchain-extension/extension/extension.ts

```diff
@@ -116,8 +124,30 @@ export async function activate(context: vscode.ExtensionContext): Promise<void>
         const tempCommandRegistry: TemporaryCommandRegistry = TemporaryCommandRegistry.instance();
 
         // Register the 'Open Pre Req' command
-        context = await ExtensionUtil.registerPreReqAndReleaseNotesCommand(context);
-
+        context = await ExtensionUtil.registerPreActivationCommands(context);
+
+        if (extensionUpgraded) {
+            // If user upgraded major versions, if old local ansible environmens present ask if they should be teardown
+            // This needs to happen before rendering pannels, otherwise the user will see the old environments
+            const v1ExtDir: string = extDir.replace(`${path.sep}v2`, '');
+            FabricEnvironmentRegistry.instance().setRegistryPath(v1ExtDir);
+            try {
+                const v1EnvironmentEntries: FabricEnvironmentRegistryEntry[] = await FabricEnvironmentRegistry.instance().getAll();
+                if (v1EnvironmentEntries.length > 0) {
+                    const teardown: any = await UserInputUtil.showConfirmationWarningMessage('Detected old local environments which can not be used with this upgrade. Would you like to teardown these environments?');
+                    if (teardown) {
+                        for (const entry of v1EnvironmentEntries) {
+                            await vscode.commands.executeCommand(ExtensionCommands.TEARDOWN_FABRIC, undefined, true, entry.name, true);
```

Comment:
I think we should try/catch each teardown (can ignore the error), just so this won't potentially stop the entire extension.

Reply:
It wouldn't stop the extension, but I'm happy to try/catch each.